### PR TITLE
feat(map)

### PR DESCRIPTION
**Pull request by sandymash**
- added map component with location autocomplete input
- added google maps types
- geocode parsing is now made on location select and not on save
- divided geocoder and autocomplete service hooks
- removed disable-clearable on autocomplete fields

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -36,9 +37,11 @@
         "dotenv": "^8.2.0",
         "enzyme": "^3.11.0",
         "enzyme-adapter-react-16": "^1.15.3",
+        "google-map-react": "^2.1.5",
         "jss-rtl": "^0.3.0",
         "react": "^16.13.1",
         "react-dom": "^16.13.1",
+        "react-google-maps": "^9.4.5",
```
Comment (sandymash) [inline client/package.json:44]: remove

File: client/src/Utils/ControllerHooks/useExposuresSaving.ts [hunk 2]
```diff
@@ -37,7 +36,7 @@ const useExposuresSaving = (exposuresAndFlightsVariables: ExposureAndFlightsDeta
             exposureAndDataToReturn = {
                 ...exposureAndDataToReturn,
                 // @ts-ignore
```
Comment (sandymash) [inline client/src/Utils/ControllerHooks/useExposuresSaving.ts:38]: add different type for return

File: client/src/Utils/vendor/useDBParsing.ts [hunk 3]
```diff
@@ -1,20 +1,16 @@
 import useGoogleApiAutocomplete from "commons/LocationInputField/useGoogleApiAutocomplete";
-import {GoogleApiPlace} from "commons/LocationInputField/LocationInput";
+import {GeocodeResponse, GoogleApiPlace} from "commons/LocationInputField/LocationInput";
 
-const useDBParser = () => {
-    const {parseAddress, requestDetailsFromPlaceId} = useGoogleApiAutocomplete();
+type parseAddressStringified = (address: (GoogleApiPlace|null), rawData: false | undefined) => Promise<string|null>;
+type parseAddressRaw = (address: GoogleApiPlace, rawData: true) => Promise<GeocodeResponse>;
+// type parseNullAddress = (address:null, rawData: boolean) => Promise<null>;
```
Comment (sandymash) [inline client/src/Utils/vendor/useDBParsing.ts:6]: rmv

File: client/src/commons/Map/Map.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    GoogleApiPlace
+} from "../LocationInputField/LocationInput";
+
+import useGoogleApiAutocomplete from "commons/LocationInputField/useGoogleApiAutocomplete";
+import injectScript from "commons/LocationInputField/useGoogleScriptInjector";
+import useGoogleGeocoder from "commons/LocationInputField/useGoogleGeocoder";
+import useDBParser from "Utils/vendor/useDBParsing";
+
+interface MapProps {
+    height?: number | string;
```
Comment (daniel7764) [inline client/src/commons/Map/Map.tsx:15]: I see that in the two times you're calling the map component you don't provide the height and width properties, so do you really need them? or there'll be in the future instances that we would want to provide our height/width?

Comment (daniel7764) [reply client/src/commons/Map/Map.tsx:15]: Also why number | string? I mean i get why string for px, vh, vw etc..

File: client/src/commons/Map/Map.tsx [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+    const handleApiLoaded = (loadObject: any) => {
+        setMap(loadObject.map)
+    };
+
+    const handleAddressSelected = (event:React.ChangeEvent<{}>, placeData: GoogleApiPlace | null) => {
+        setSelectedAddress(placeData as GeocodeResponse);
+    };
+
+    const height = props.height || '40vh';
```
Comment (daniel7764) [inline client/src/commons/Map/Map.tsx:118]: I would put the default height/width in a const

File: client/src/commons/Map/Map.tsx [hunk 6]
```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+// @ts-ignore
```
[the remark this answers is not in the document]
Comment (sandymash) [reply client/src/commons/Map/Map.tsx:2]: imported modules doesnt have types

File: client/src/commons/Map/Map.tsx (hunk 4 above)
Comment (sandymash) [reply client/src/commons/Map/Map.tsx:15]: number defaults to pxs

Comment (sandymash) [reply client/src/commons/Map/Map.tsx:15]: its added so the container size could be customizable

**Review by daniel7764 — APPROVED** [review]

**Review by daniel7764 — APPROVED** [review]

File: client/src/commons/LocationInputField/useGoogleApiAutocomplete.ts [hunk 7]
```diff
@@ -9,27 +9,12 @@ interface APIRequestParams {
 }
 
 const useGoogleApiAutocomplete = () => {
-    const [autocompleteService, setAutoCompleteService] = React.useState();
-    const [geocoder, setGeocoder] = React.useState();
+    const [autocompleteService, setAutoCompleteService] = React.useState<google.maps.places.AutocompleteService>();
```
Comment (OrelKaravani) [inline client/src/commons/LocationInputField/useGoogleApiAutocomplete.ts:12]: Use camel letters.. "autoCompleteService".

File: client/src/Utils/vendor/useDBParsing.ts [hunk 8]
```diff
@@ -1,17 +1,27 @@
 import useGoogleApiAutocomplete from "commons/LocationInputField/useGoogleApiAutocomplete";
-import {GoogleApiPlace} from "commons/LocationInputField/LocationInput";
+import {GeocodeResponse, GoogleApiPlace} from "commons/LocationInputField/LocationInput";
+import useGoogleGeocoder from "../../commons/LocationInputField/useGoogleGeocoder";
```
Comment (MayaSasson) [inline client/src/Utils/vendor/useDBParsing.ts:3]: make this an absulote path and single quotes

**Review by MayaSasson — APPROVED** [review]

**Review by OrelKaravani — APPROVED** [review]

**Review by daniel7764 — APPROVED** [review]

**Pull request merged**